### PR TITLE
[ARO-7181] Upgrade the ARO Operator, then update the Cluster object

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes ARO-7181

### What this PR does / why we need it:

Fixes a potential bug (and something that affects #6660) where a flag change *and* an ARO Operator upgrade would cause the flag to take effect on the old operator, rather than the new one.

### Test plan for issue:

Unit tests should cover the AdminUpdate behaviour, E2E tests should cover the installation behaviour. It will also undergo QA as a part of #6660.

### Is there any documentation that needs to be updated for this PR?
N/A

### How do you know this will function as expected in production? 

Testing :)

### CODE REVIEW

File: pkg/operator/deploy/deploy.go [hunk 1]
```diff
@@ -277,31 +301,33 @@ func (o *operator) resources(ctx context.Context) ([]kruntime.Object, error) {
...
+func (o *operator) SyncClusterObject(ctx context.Context) error {
+	resource, err := o.clusterObject(ctx)
+	if err != nil {
+		return err
+	}
+	return o.dh.Ensure(ctx, resource)
 }
 
-func (o *operator) CreateOrUpdate(ctx context.Context) error {
+func (o *operator) CreateOrUpdate(ctx context.Context, isInstall bool) error {
```
Comment (mociarain) [inline pkg/operator/deploy/deploy.go:315]: I think we should split this into 2 separate functions: Create and Update with a common internal method.

File: pkg/cluster/adminupdate_test.go [hunk 2]
```diff
@@ -56,6 +56,7 @@ func TestAdminUpdateSteps(t *testing.T) {
 				"[Action ensureAROOperator-fm]",
 				"[Condition aroDeploymentReady-fm, timeout 20m0s]",
 				"[Condition ensureAROOperatorRunningDesiredVersion-fm, timeout 5m0s]",
+				"[Action syncClusterObject-fm]",
```
Comment (mociarain) [inline pkg/cluster/adminupdate_test.go:59]: I think it's worth adding a test for just the `MaintenanceTaskSyncClusterObject=SyncClusterObject`.

File: pkg/operator/deploy/deploy.go [hunk 3]
```diff
@@ -207,6 +208,29 @@ func (o *operator) resources(ctx context.Context) ([]kruntime.Object, error) {
 		return nil, err
 	}
 
+	domain := o.oc.Properties.ClusterProfile.Domain
+	if !strings.ContainsRune(domain, '.') {
+		domain += "." + o.env.Domain()
+	}
```
Comment (mociarain) [inline pkg/operator/deploy/deploy.go:214]: Why is this needed?

**Review by mociarain — CHANGES_REQUESTED** [review]
Few questions and suggestions

File: pkg/cluster/adminupdate_test.go (hunk 2 above)
Comment (hawkowl) [reply pkg/cluster/adminupdate_test.go]: done

File: pkg/operator/deploy/deploy.go (hunk 1 above)
Comment (hawkowl) [reply pkg/operator/deploy/deploy.go]: done

File: pkg/operator/deploy/deploy.go (hunk 3 above)
Comment (hawkowl) [reply pkg/operator/deploy/deploy.go]: incorrect copy-paste, fixed

**Review by mociarain — DISMISSED** [review]
Nice. Looks great

**Review by jaitaiwan — APPROVED** [review]
One small nit, don't think it needs actioning in this PR. Approving and merging
